libs/blocks/comparison-table/comparison-table.js

```diff
@@ -0,0 +1,529 @@
+    if (col.classList.contains('header-item') && firstHeaderItem !== col) {
+      parent.insertBefore(col, firstHeaderItem.nextSibling);
+      return;
+    }
+    const rowHeader = parent.querySelector('.table-row-header');
+    if (rowHeader) parent.insertBefore(col, rowHeader.nextSibling);
+  });
+
+  const selectElement = el.querySelector(`[data-column-index="${newValue}"] .mobile-filter-select`);
+  if (selectElement) selectElement.value = newValue;
+  updateVisibleSelects({ el, headerTitles });
+  syncAccessibilityHeaders(el);
+}
+
+function createMobileFilterSelect({ headerTitles, headerItemIndex, el }) {
+  const select = createTag('select', { class: 'mobile-filter-select', name: 'column-filter' });
+  headerTitles.forEach((title, index) => {
+    if (!title
+      || (headerItemIndex === 1 && index === 2)
+      || (headerItemIndex === 2 && index === 1)) return;
+    const option = createTag('option', { value: index }, title);
+    if (index === headerItemIndex) option.selected = true;
+    select.appendChild(option);
+  });
+  select.addEventListener('change', (e) => handleSelectChange(e, { headerItemIndex, el, headerTitles }));
+  return select;
+}
+
+function addLastContainerElements(container) {
+  const actionAreaElements = container.querySelectorAll('.action-area');
+  if (actionAreaElements.length) {
+    const btnContainer = createTag('div', { class: 'btn-container' });
+    if (actionAreaElements.length > 1) btnContainer.classList.add('has-multiple');
+    btnContainer.append(...actionAreaElements);
+    container.appendChild(btnContainer);
+  }
+  let description = container.querySelector('p:not(.action-area)');
+  if (!description) {
+    description = createTag('p', { class: 'description' });
+    container.prepend(description);
+    return;
+  }
+  description.classList.add('description');
+}
+
+function createSubHeaderContainer({
+  childrenArray,
+  startIndex,
+  endIndex,
+  el,
+  isLast = false,
+  isFirst = false,
+  headerTitles = [],
+  headerItemIndex = 0,
+  headerItemsCount = 0,
+}) {
+  const container = createTag('div', { class: 'sub-header-item-container' });
+  const decoratePromises = [];
+
+  for (let i = startIndex; i < endIndex; i += 1) {
+    if (childrenArray[i] && childrenArray[i].textContent.trim() !== '-') {
+      container.appendChild(childrenArray[i]);
+      if (isLast && childrenArray[i].querySelector('strong')) {
+        const promise = import('../../utils/decorate.js').then(({ decorateButtons }) => {
+          decorateButtons(childrenArray[i]);
+        });
+        decoratePromises.push(promise);
+      }
+    }
+  }
+  if (isFirst && headerItemsCount > 3) {
+    const select = createMobileFilterSelect({ headerTitles, headerItemIndex, el });
+    container.appendChild(select);
+  }
+  if (!isLast) return container;
+  if (decoratePromises.length) {
+    Promise.all(decoratePromises).then(() => addLastContainerElements(container));
+  } else {
+    addLastContainerElements(container);
+  }
```

Comment:
Nit: You should be able to just pass `decoratePromises` to `Promise.all`. `Promise.all([]) => []`. And question, does this need to be awaited ? I see that `addLastContainerElements` adds an element, if promises resolve late, can there be a CLS if element is added later ? 
```suggestion
   Promise.all(decoratePromises).then(() => addLastContainerElements(container));
```